File: Python Course/Software Development Module - in Python/Week11/Project.py
"""
Project.py
Billy Cussen
06/10/2020
"""
import Assessment

class myProject(Assessment.myAssessment):

    def __init__(self, weighting, name, typeOfAssessment, deadline):
        super().__init__(weighting, name, typeOfAssessment)
        self.deadline = deadline

    def setDeadline(self, deadline):
        self.deadline = deadline

    def getDeadline(self):
        return self.deadline

    def toString(self):
        return "Weighting: "+str(self.weighting)+"\nName: "+self.name+"\nType of Assessment: "+self.typeOfAssessment+"\nDeadline: " + str(self.deadline)
        

File: Java/Software Development Module - Java/Week11/Project.java
/*
Project.java
Billy Cussen
21/07/2020
*/

import java.util.Date;

public class Project extends Assessment{

    //Data Members
    private Date deadline;

    //Constructors
    public Project(){
        this.deadline = new Date();
    }

    public Project(int weighting, String name, String type, Date deadline){
        super(weighting, name, type);
        this.deadline = deadline;
    }



    //Setter
    public void setDate(Date deadline){
        this.deadline = deadline;
    }

    //Getter
    public Date getDate(){
        return this.deadline;
    }

    @Override
    public String toString(){
        return "Name: "+this.name+", Weighting: "+this.weighting+", Type: "+this.type+", Deadline: "+this.deadline;
    }

}